File: epi_judge_java_solutions/epi/IsNumberPalindromic.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class IsNumberPalindromic {

  @EpiTest(testDataFile = "is_number_palindromic.tsv")
  public static boolean isPalindromeNumber(int x) {

    if (x <= 0) {
      return x == 0;
    }

    final int numDigits = (int)(Math.floor(Math.log10(x))) + 1;
    int msdMask = (int)Math.pow(10, numDigits - 1);
    for (int i = 0; i < (numDigits / 2); ++i) {
      if (x / msdMask != x % 10) {
        return false;
      }
      x %= msdMask; // Remove the most significant digit of x.
      x /= 10;      // Remove the least significant digit of x.
      msdMask /= 100;
    }
    return true;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "IsNumberPalindromic.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi_judge_python_solutions/is_number_palindromic.py
import math

from test_framework import generic_test


def is_palindrome_number(x: int) -> bool:

    if x <= 0:
        return x == 0

    num_digits = math.floor(math.log10(x)) + 1
    msd_mask = 10**(num_digits - 1)
    for i in range(num_digits // 2):
        if x // msd_mask != x % 10:
            return False
        x %= msd_mask  # Remove the most significant digit of x.
        x //= 10  # Remove the least significant digit of x.
        msd_mask //= 100
    return True


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('is_number_palindromic.py',
                                       'is_number_palindromic.tsv',
                                       is_palindrome_number))
